activejob/lib/active_job/test_helper.rb

```diff
@@ -39,10 +39,10 @@ def enable_test_adapter(test_adapter)
     end
 
     def before_setup # :nodoc:
-      test_adapter = queue_adapter_for_test
-
       queue_adapter_changed_jobs.each do |klass|
-        klass.enable_test_adapter(test_adapter)
+        if (test_adapter = test_adapter_should_be(klass))
```

Comment:
Not a big fan of that `test_adapter_should_be` name, I'm really struggling a bit too much to understand what it does.

Maybe it would make more sense to just inline it here. Or to move the responsability to call `enable_test_adapter` there and rename it. But I'd do the former.